engines/mock/mocksandbox.go

```diff
@@ -70,13 +74,29 @@ func (s *sandbox) AttachProxy(name string, handler http.Handler) error {
 	// Lock before we access proxies as this method may be called concurrently
 	s.Lock()
 	defer s.Unlock()
+	if strings.ContainsAny(name, " ") {
+		return engines.NewMalformedPayloadError("MockEngine proxy names cannot contain space")
+	}
```

Comment:
Would be better to include the proxy name that was used in the error.

`return engines.NewMalformedPayloadError(fmt.Sprintf("MockEngine proxy name %q should not contain space", name))`